cubano-webdriver/src/test/java/org/concordion/cubano/driver/web/config/WebDriverConfigTest.java

```diff
@@ -1,177 +1,166 @@
-        assertThat(config.getBrowserExe(), is("/bin/firefox"));
-        assertThat(config.shouldActivatePlugins(), is(true));
-    }
-
-    @Test
-    public void implicitWaitDefaultsToZero() {
-        assertThat(System.getProperty("webdriver.timeouts.implicitlywait"), is(nullValue()));
-
-        Properties properties = givenDefaultProperties();
-        WebDriverConfig config = new WebDriverConfig(properties);
-
-        assertThat(System.getProperty("webdriver.timeouts.implicitlywait"), is("0"));
-    }
-
-
-    @Test
-    public void implicitWaitCanBeSet() {
-        assertThat(System.getProperty("webdriver.timeouts.implicitlywait"), is(nullValue()));
-
-        Properties properties = givenDefaultProperties();
-        given(properties.getProperty("webdriver.timeouts.implicitlywait")).willReturn("9");
-
-        WebDriverConfig config = new WebDriverConfig(properties);
-
-        assertThat(System.getProperty("webdriver.timeouts.implicitlywait"), is("9"));
-    }
-
-    private Properties givenDefaultProperties() {
-        Properties properties = mock(Properties.class);
-        given(properties.getProperty("environment")).willReturn("UAT");
-        given(properties.getProperty("webdriver.browser")).willReturn("firefox");
-        given(properties.getProperty("webdriver.defaultTimeout")).willReturn("30");
-        return properties;
-    }
-
+package org.concordion.cubano.driver.web.config;
+
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.RestoreSystemProperties;
+
+import java.util.Properties;
+
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.CoreMatchers.nullValue;
+import static org.junit.Assert.*;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.mock;
+
+public class WebDriverConfigTest {
+    @Rule
+    public final RestoreSystemProperties restoreSystemProperties
+            = new RestoreSystemProperties();
+
+    @Test
+    public void mustSetDefaultProperties() throws Exception {
+        Properties properties = mock(Properties.class);
+        given(properties.getProperty("environment")).willReturn("UAT");
+        given(properties.getProperty("webdriver.browser")).willReturn("firefox");
+        given(properties.getProperty("webdriver.defaultTimeout")).willReturn("30");
+
+        WebDriverConfig config = new WebDriverConfig(properties);
+
+        assertThat(config.getEnvironment(), is("UAT"));
+        assertThat(config.getBrowser(), is("firefox"));
+        assertThat(config.getDefaultTimeout(), is(30));
+        assertThat(config.isProxyRequired(), is(false));
+    }
+
+    @Test
+    public void canSetProxyProperties() {
+        Properties properties = givenDefaultProperties();
+        given(properties.getProperty("proxy.required")).willReturn("true");
+        given(properties.getProperty("proxy.host")).willReturn("myproxyhost:9999");
+        given(properties.getProperty("proxy.username")).willReturn("mydomain\\me");

```

Comment:
Is this Windows specific? Or can unix proxies use this format?

Reply:
A unix proxy would never specify the domain, a windows proxy doesn't need to either unless need to force ntlm authentication rather than basic username password auth.